lib/rules/switch-final-break.js

```diff
@@ -0,0 +1,158 @@
+         */
+        function isReachable(segment) {
+            return segment.reachable;
+        }
+
+
+        return {
+            onCodePathStart(codePath) {
+                currentCodePath = codePath;
+            },
+            onCodePathEnd() {
+                currentCodePath = currentCodePath.upper;
+            },
+
+            "SwitchCase:exit"(node) {
+                if (requireBreak !== "always") {
+                    return;
+                }
+                if (node.consequent.length === 0) {
+                    context.report({
+                        node,
+                        messageId: "switchFinalBreakAlways",
+                        fix: fixer => fixer.insertTextAfter(
+                            node,
+                            " break;"
+                        )
+                    });
+                }
+                if (currentCodePath.currentSegments.some(isReachable) &&
+                    node.consequent.length > 0 &&
+                    lodash.last(node.parent.cases) === node
+                ) {
+                    const lastStatement = getLastStatement(node);
+
+                    context.report({
+                        node,
+                        messageId: "switchFinalBreakAlways",
+                        fix: fixer => fixer.insertTextAfter(
+                            lastStatement,
+                            " break;"
+                        )
+                    });
+                }
+            },
+
+            SwitchStatement(node) {
+                if (node.cases.length !== 0) {
+                    const lastCase = lodash.last(node.cases);
+                    const lastStatement = getLastStatement(lastCase);
+
+                    if (requireBreak === "always") {
+                        return;
+                    }
+
+                    if (
+                        !lastStatement ||
+                        lastStatement.type !== "BreakStatement"
+                    ) {
+                        return;
+                    }
+
+                    if (lastStatement.label !== null) {
+                        const parent = node.parent;
+
+                        if (
+                            parent &&
+                            (!(parent.type === "LabeledStatement") ||
+                                parent.label === lastStatement.label)) {
+
+                            // break is used to jump to somewhere else, don't complain. This happens when the code has a break statement with a labeled block ("break labelName;").
+                            return;
+                        }
+                    }
+                    context.report({
+                        node: lastStatement,
+                        messageId: "switchFinalBreakNever",
+                        fix: fixer => fixer.removeRange([
+                            lastStatement.range[0],
+                            lastStatement.range[1]
+                        ])
```

Comment:
I think this can be simplified with  `fixer.remove(lastStatement)` instead of `removeRange`.